src/Fantomas/Trivia.fs

```diff
@@ -253,6 +253,9 @@ let private addTriviaToTriviaNode (startOfSourceCode: int) (triviaNodes: TriviaN
 
         let nodeBefore =
             findNodeBeforeLineAndColumn triviaNodes range.StartLine range.StartColumn
+            |> function
```

Comment:
I'm a bit afraid we will try too hard to find a `nodeBefore` and do too many lookups by this.
Trivia is the real performance killer in large files, so we need to be cautious.

I would be more in favour of only attempting this when both nodes are None.
```fsharp
        | None, None ->
            findNodeBeforeLineFromStart triviaNodes range.StartLine
            |> updateTriviaNode (fun tn -> tn.ContentAfter.Add(Comment(BlockComment(comment, true, true)))) triviaNodes
```